### PR TITLE
[MH2018-14] --> [MH-Team-2-Shopify] (2) Feature/related products shopify

### PR DESCRIPTION
**Pull request by luisarmandodv**
# Task: [MH2018-14]

### Mobile:

captura de pantalla 2018-03-27 a la s 16 21 49

### Desktop:

captura de pantalla 2018-03-27 a la s 16 29 37

### Mobile:

captura de pantalla 2018-03-27 a la s 16 21 03

### Desktop:

captura de pantalla 2018-03-27 a la s 16 28 59

### CODE REVIEW

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    .product-card-category {
+
+      &:last-child {
+
+        @media only screen and (max-width : 1139px) and (min-width: 769px) {
+          display: none;
+        }
+      }
+
+      &:nth-last-child(2) {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 3

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      }
+
+      @media #{$breakpoint-tablet} {
+        justify-content: center;
+      }
+    }
+
+    .product-card-category {
+
+      &:last-child {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 3

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          }
+        }
+      }
+
+      @media #{$breakpoint-tablet} {
+        justify-content: center;
+      }
+    }
+
+    .product-card-category {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 3

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  padding-right: 30px;
+
+  .collection-container {
+    background-color: #{$product-card-category};
+
+    .product-image {
+      justify-content: space-between;
+
+
+      a {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 4

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 5]
```diff
@@ -0,0 +1,73 @@
+.related-product-container {
+  background-color: #{$product-card-category};
+  padding-left: 30px;
+  padding-right: 30px;
+
+  .collection-container {
+    background-color: #{$product-card-category};
+
+    .product-image {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 3

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      justify-content: space-between;
+
+
+      a {
+        &:nth-of-type(odd) {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Nesting should be no greater than 4, but was 5<br>Selector should have depth of applicability no greater than 2, but was 4

File: Lil-Peep/src/styles/theme.scss [hunk 7]
```diff
@@ -30,5 +30,6 @@
 @import url('modules/site-footer.scss');
 @import url('modules/pdp.scss');
 @import url('modules/gift-card-template.scss');
+@import url('modules/related-products.scss');
 @import url('modules/gift-card-template.scss');
-@import url('modules/featured-categories.scss');
+@import url('modules/featured-categories.scss');
```
Comment (magmahound) [inline Lil-Peep/src/styles/theme.scss]: Files should end with a trailing newline

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 8]
```diff
@@ -0,0 +1,78 @@
+<div class="collection-container">
+  {% for product in collection.products %}
+    <div class="product-card-category">
+      <div class="products-slider">
+        <div class="front-label">
+        {% assign date_created = product.created_at | date:'%s' %}
+        {% assign     date_now = 'now' | date:'%s' %}
+        {% assign   difference = date_now | minus: date_created %}
```
Comment (albertm805) [inline Lil-Peep/src/snippets/collection-container.liquid]: Whattup with the space? is this supposed to be like this?

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  background-color: #{$product-card-category};
+  padding-left: 30px;
+  padding-right: 30px;
+
+  .collection-container {
+    background-color: #{$product-card-category};
+
+    .product-image {
+      justify-content: space-between;
+
```
Comment (albertm805) [inline Lil-Peep/src/styles/modules/related-products.scss]: Double spacing?

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+            {% if product.compare_at_price > product.price %}
+              <span class="visually-hidden">{{ 'products.product.regular_price' | t }}</span>
+              <p class="offer">{{ product.compare_at_price | money }} USD </p>
+            {% endif %}
+
+            {% unless product.available %}
+              {{ 'products.product.sold_out' | t }}
+            {% endunless %}
+
```
Comment (rubiocanino) [inline Lil-Peep/src/snippets/collection-container.liquid]: Unnecessary space

Comment (rubiocanino) [inline Lil-Peep/src/snippets/collection-container.liquid]: Unnecessary space

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+          {% endif %}
+        </div>
+      </div>
+      <div class="product-data">
+        <p>
+          <a href="{{ product.url | within: collection }}" class="product-title">{{ product.title }}</a>
+        </p>
+        <div class="price">
+          <p class="price">
+
```
Comment (rubiocanino) [inline Lil-Peep/src/snippets/collection-container.liquid]: Unnecessary space

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+            <h4>{{ 'collections.general.view' | t }}</h4>
+          </a>
+        </div>
+      </div>
+    </div>
+
+  {% else %}
+    {% if collection.handle == 'all' and collection.all_vendors.size == 0 and collection.all_types.size == 0 %}
+      {% for i in (1..8) %}
+        <a href="#">
```
Comment (IsmaelVillavicencio) [inline Lil-Peep/src/snippets/collection-container.liquid]: You should use `<a href="javascript:void(0)">`

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+          {% capture current %}{% cycle 1, 2, 3, 4, 5, 6 %}{% endcapture %}
+          {{ 'product-' | append: current | placeholder_svg_tag: 'placeholder-svg placeholder-svg--small' }}
+        </a>
+
+        <p>
+          <a href="#">{{ 'homepage.onboarding.product_title' | t }}</a>
```
Comment (IsmaelVillavicencio) [inline Lil-Peep/src/snippets/collection-container.liquid]: You should use `<a href="javascript:void(0)">`

File: Lil-Peep/src/sections/product.liquid [hunk 14]
```diff
@@ -128,3 +128,20 @@
     </script>
   {% endunless %}
 </div>
+
+<div class="related-product-container">
+  <h2 class="related-product-title">YOU MAY ALSO LIKE</h2>
```
Comment (IsmaelVillavicencio) [inline Lil-Peep/src/sections/product.liquid]: You should use `Locales`

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+  {% for product in collection.products %}
+    <div class="product-card-category">
+      <div class="products-slider">
+        <div class="front-label">
+        {% assign date_created = product.created_at | date:'%s' %}
+        {% assign     date_now = 'now' | date:'%s' %}
+        {% assign   difference = date_now | minus: date_created %}
+
+        {% if difference < 172800 %}
+          <p class="new-product">NEW</p>
```
Comment (IsmaelVillavicencio) [inline Lil-Peep/src/snippets/collection-container.liquid]: You should use `locales`

File: Lil-Peep/src/snippets/collection-container.liquid [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+        {% else %}
+          <p class="not-new-product">NOT NEW</p>
```
Comment (IsmaelVillavicencio) [inline Lil-Peep/src/snippets/collection-container.liquid]: You should use `locales`

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 17]
```diff
@@ -0,0 +1,73 @@
+.related-product-container {
+  background-color: #{$product-card-category};
```
Comment (jaimelr) [inline Lil-Peep/src/styles/modules/related-products.scss]: This also works like `background-color: $product-card-category`, it's cleaner and easier to read.

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+  padding-left: 30px;
+  padding-right: 30px;
+
+  .collection-container {
+    background-color: #{$product-card-category};
```
Comment (jaimelr) [inline Lil-Peep/src/styles/modules/related-products.scss]: This also works like `background-color: $product-card-category;`, it's cleaner and easier to read.

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+
+  @media #{$breakpoint-tablet} {
+    background-color: #{$background-color};
+  }
+}
+
+.related-product-title {
+  color: #{$title-font-gray};
```
Comment (jaimelr) [inline Lil-Peep/src/styles/modules/related-products.scss]: This also works like `color: $title-font-gray;`, it's cleaner and easier to read.

Comment (luisarmandodv) [reply Lil-Peep/src/styles/modules/related-products.scss]: really?! I did know that! I feel cheated.

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+  padding-left: 30px;
+  padding-right: 30px;
+
+  .collection-container {
+    background-color: $product-card-category;
+
+    .product-image {
+      justify-content: space-between;
+
+      a {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Selector should have depth of applicability no greater than 2, but was 4

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+        &:nth-of-type(odd) {
```
Comment (magmahound) [inline Lil-Peep/src/styles/modules/related-products.scss]: Nesting should be no greater than 4, but was 5<br>Selector should have depth of applicability no greater than 2, but was 4

File: Lil-Peep/src/sections/featured-product.liquid [hunk 22]
```diff
@@ -8,143 +8,20 @@
...
-  </div>
-
-  <div class="rte">
-    {{ product.description }}
+  {%- assign collection_product = collections.featured_products -%}
+  <div class="related-product-container">
+    <h2 class="related-product-title">{{ 'homepage.onboarding.featured-products' | t }}</h2>
+    <div class="mobile">
+      {% paginate collection_product.products by 1 %}
+        {%- include 'collection-container', product_collection: collection_product -%}
```
Comment (rubiocanino) [inline Lil-Peep/src/sections/featured-product.liquid]: - or _ ?

File: Lil-Peep/src/sections/featured-product.liquid [hunk 23]
```diff
@@ -8,143 +8,20 @@
...
+      {% endpaginate %}
+    </div>
+
+    <div class="desktop">
+      {% paginate collection_product.products by 4 %}
+        {%- include 'collection-container', product_collection: collection_product -%}
```
Comment (rubiocanino) [inline Lil-Peep/src/sections/featured-product.liquid]: - or _  ?

File: Lil-Peep/src/styles/modules/related-products.scss [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+  font-weight: 100;
+  padding-top: 20px;
+  text-align: center;
+
+  @media #{$breakpoint-tablet} {
+    margin-left: 10px;
+    text-align: left;
+
+    &:after {
+      content: ":";
```
Comment (8geonirt) [inline Lil-Peep/src/styles/modules/related-products.scss]: Is this necessary? 🤔

Comment (luisarmandodv) [reply Lil-Peep/src/styles/modules/related-products.scss]: yep, look at this:

captura de pantalla 2018-03-29 a la s 11 47 23
captura de pantalla 2018-03-29 a la s 11 47 32

Comment (luisarmandodv) [reply Lil-Peep/src/styles/modules/related-products.scss]: differences between mobile and desktop versions 🤔

Comment (8geonirt) [reply Lil-Peep/src/styles/modules/related-products.scss]: Oh, I see, looks like this was a design error, but yeah, you accomplished the design, that's nice :)

**Pull request merged**